### DISCUSSION TITLE
Can you change the web GUI port from 5000?

### DISCUSSION THREAD

**Discussion by pdjohntony** [blakeblackshear/frigate#3032] [Ask A Question]
I am trying out Frigate in Docker on my Synology NAS. Its running but I can't access the GUI due to Synology itself using port 5000 for the HTTP redirect to HTTPS. Is it possible to change this port in Frigate?

**Comment by stephengolub** [comment]
Ensure you're not running the container in "host" mode and then map the container port `5000` to a different port on the host. Such as `8080` or `6000`.

Make sure this is unchecked on the "Network" tab when creating the container (by default it is):

<img width="305" alt="image" src="https://user-images.githubusercontent.com/1036328/160869915-394ad640-f270-43a9-85b2-924da384726d.png">

Then go to port settings and change "Auto" to something else:
<img width="628" alt="image" src="https://user-images.githubusercontent.com/1036328/160870111-6c829490-bcf5-4b70-8b47-eeae759f3c46.png">

**Reply by pdjohntony** [reply]
Network is bridge mode and I already tried mapping to port 5002.
![image](https://user-images.githubusercontent.com/6294702/160871302-2fd6aeae-a420-4da8-855c-8a6236d807f4.png)
![image](https://user-images.githubusercontent.com/6294702/160871235-6e7d76bc-fde6-4bce-ae70-10f6f889146b.png)

I've also made sure port 5002 is allowed through the Synology firewall. The port appears open, its just that Frigate isn't using it.

**Reply by stephengolub** [reply]
Ah, don't change the container port. You're mapping the host to the port inside the container.

**Reply by pdjohntony** [reply]
I just found the Synology DSM port settings and changed HTTP from port 5000 to 4999. Then I was able to change the container port to 5000 and now I can access Frigate.

![image](https://user-images.githubusercontent.com/6294702/160872836-1e0cd496-3f2d-44d2-86b4-90b476923e79.png)

This still doesn't answer my original question if Frigate's GUI port is changeable itself, but at least I got it working for now.

**Reply by NickM-27** [reply]
Frigate does not need the port changed, each docker container is an internal network and has no reason to need a different port. As @stephengolub suggested leaving container port as 5000 and changing the local port would have also worked just fine.

**Reply by stephengolub** [reply]
Yea, I'd suggest reading up on docker containers and port mapping on network so you can avoid reconfiguring the host machine in the future.

**Reply by NickM-27** [reply]
Just as another example, here is docker on Unraid. You can see multiple containers all use port 5000 and yet my host has different ports defined for them. When the container (like Frigate) accepts a port that is something that is hardwired in and can't by dynamically changed. This is a mandatory feature for any docker implementation.

<img width="316" alt="Screen Shot 2022-03-30 at 9 34 30 AM" src="https://user-images.githubusercontent.com/14866235/160874368-ac5eebc7-3811-4082-93e0-1c22365296de.png">

**Reply by pdjohntony** [reply]
Right, I'm still very new to docker but I do understand that each container has its own internal network. I know a container can still use port 5000 since that's just internally inside its own container network. Where I am hung up is if the host is already using port 5000 and if I wish to expose the container's port 5000, so other computers on the network could access it, there would be a conflict. Unless docker has some sort of PAT configuration that I am not seeing.

**Reply by NickM-27** [reply]
@pdjohntony I don't think you 100% understand, the network ports do not need to be the same (see above). You can easily map port `6000` of your host computer to port `5000` of frigate container. Then simply access `your.pc.ip.address:6000` to get to frigate from any device.

**Reply by stephengolub** [reply]
Basically you don't have to map the host port to the same port as the container.

**Reply by pdjohntony** [reply]
I just saw your last post with the Unraid screenshot. No I do understand that. So that would be Port Address Translation (PAT). Now I just need to figure out how to do that in Docker/Portainer.

Thank you very much for the replies @stephengolub @NickM-27. I appreciate the help.

**Reply by NickM-27** [reply]
@pdjohntony Right, Unraid uses docker too, it is a mandatory Docker feature. The portainer docs show how to do it here: https://docs.portainer.io/v/ce-2.11/user/docker/services/configure#published-ports
